src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/Deflater.cs

```diff
@@ -134,9 +134,12 @@ internal int GetDeflateOutput(byte[] outputBuffer)
 
         private unsafe ZErrorCode ReadDeflateOutput(byte[] outputBuffer, ZFlushCode flushCode, out int bytesRead)
         {
+            Debug.Assert(null != outputBuffer, "Can't pass in a null output buffer!");
+            Debug.Assert(outputBuffer.Length > 0, "Can't pass in an empty output buffer!");
+        
             lock (SyncLock)
             {
-                fixed (byte* bufPtr = outputBuffer)
+                fixed (byte* bufPtr = &outputBuffer[0])
```

Comment:
We use asserts in places like this (internal/private members) as a way of saying "our own callers must have already verified these conditions are met."  Without this change, if a null outputBuffer is passed in, you're going to get a null ref two lines later when we try to access .Length, so this change would simply cause that null ref to happen two lines earlier.  For an empty outputBuffer, there would be a change, as before we'd be telling zlib there was 0 output space available, at which point the deflate call would return an error indicating lack of buffer space, and with this change, you'd get an out of range exception from trying to index outputBuffer.  But all of the call sites to this pass in a non-empty buffer, meeting the internal contract being spelled out by the asserts.